src/utils/pipeSpawn.js

```diff
@@ -1,16 +1,26 @@
 const spawn = require('cross-spawn');
+const logger = require('../Logger');
 
 function pipeSpawn(cmd, params, opts) {
-  const cp = spawn(cmd, params, opts);
-  cp.stdout.pipe(process.stdout);
-  cp.stderr.pipe(process.stderr);
+  const cp = spawn(cmd, params, Object.assign({
+    env: Object.assign({
+      FORCE_COLOR: logger.color,
+      npm_config_color: logger.color ? 'always': '',
+      npm_config_progress: true
+    }, process.env)
+  }, opts));
+
+  cp.stdout.setEncoding('utf8').on('data', d => logger.writeRaw(d));
+  cp.stderr.setEncoding('utf8').on('data', d => logger.writeRaw(d));
```

Comment:
`pipeSpawn` now uses the logger to write its output rather than directly piping to stdout. This creates a better experience when installing packages: when the install is complete, the output from yarn or npm is cleared since it is no longer relevant and the build continues. When an error occurs, it stays up to allow debugging.